akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala

```diff
@@ -255,7 +255,7 @@ object BasicPersistentBehaviorCompileOnly {
     commandHandler = (state, cmd) => throw new NotImplementedError("TODO: process the command & return an Effect"),
     eventHandler = (state, evt) => throw new NotImplementedError("TODO: process the event return the next state"))
   //#snapshottingPredicateDeleteEvents
-    .snapshotWhen(predicate, deleteEventsOnSnapshot = true)
+    .withRetention(RetentionCriteria.snapshotWhen(predicate).withDeleteEventsOnSnapshot)
```

Comment:
I like this API

Reply:
Shall we deprecate the old snapshotWhen way?